### PR TITLE
Update to signature version 4.

### LINKED ISSUE #60: Use AWS signature version 4 (required starting October 2020)
The `mime-mail-ses` package currently uses Signature Version 3, as can be seen by the string "AWS3-HTTPS" in the `X-Amzn-Authorization` header [1]. Starting 1 October 2020, AWS SES will require clients to use Signature Version 4 [2].

Unless fixed until then, the `sendMailSES` function will no longer work.

[1] https://www.stackage.org/haddock/lts-16.5/mime-mail-ses-0.4.1/src/Network.Mail.Mime.SES.html#sendMailSES
[2] https://docs.aws.amazon.com/ses/latest/DeveloperGuide/using-ses-api-authentication.html

### PR DESCRIPTION
**Pull request by kindaro**
Resolve #60.

### CODE REVIEW

**Comment by snoyberg** [comment]
Thanks @kindaro!

@limick since you opened the original issue, would you be interested in reviewing before this is merged?

File: mime-mail-ses/mime-mail-ses.cabal [hunk 1]
```diff
@@ -32,4 +32,22 @@ Library
                      , conduit
                      , cryptohash         >= 0.7.3
                      , byteable
+                     , HTTP
```
Comment (snoyberg) [inline mime-mail-ses/mime-mail-ses.cabal:35]: Maybe I'm overlooking something obvious, but I don't see where this library is being used above.

Comment (kindaro) [reply mime-mail-ses/mime-mail-ses.cabal]: [See new commit.][1]

[1]: https://github.com/snoyberg/mime-mail/pull/61/commits/6e7af5031b1184cb54165b4c3eb44622a146626b

**Comment by kindaro** [comment]
@snoyberg  I added two small patches. If it is good for you, let me know and I shall squash.

**Comment by snoyberg** [comment]
LGTM. You interested in the Hackage maintainer bit?

**Comment by kindaro** [comment]
@snoyberg  I can try my hand at it, but I may still need an advice from time to time. If with this consideration in mind you are still willing to extend your offer, then surely I shall do my best to stand up to the responsibility.

**Comment by snoyberg** [comment]
Offer is definitely on the table still, and I'd be happy to provide guidance as necessary. You should also have an invite to be maintainer here. I'd say: merge here when ready, and if you give me your Hackage username I'll give you upload permissions.

**Comment by jezen** [comment]
This perhaps warrants a version bump for Hackage.

**Comment by kindaro** [comment]
@jezen  I am working on it. Expect in about 2 days.

**Comment by jezen** [comment]
@kindaro Understood. Thank you so much for your work on this! 👏

**Pull request merged**